### DISCUSSION TITLE
Trade Fee Discussion

### DISCUSSION THREAD

**Discussion by starsoccer** [starsoccer/cryptotithe#285] [Ideas]
@Zahrun Discussion for trade fee implementation

**Comment by Zahrun** [comment]
I think here we can have different cases because exchanges are different.

1. The fee is included in the trade amount and rate in the csv
    1. We could either include it in the trade without storing the information of the fee paid, the calculation of holdings and gains stays the same
    2. Or have the trade and the fee information both stored and integrate the fee in the calculation, but make sure to not include it twice.

2. The fee is additional to the trade. There can be two cases here:
    1. tradeFeeCurrency is one of the traded currencies
        1. We could either include it in the trade without storing the information of the fee paid
        2. Or have the trade and the fee information both stored and integrate the fee in the calculation
    2. tradeFeeCurrency is a different coin
        1. tradeFee and tradeFeeCurrency have to be filled
        2. I think the difference in how calculations are done is that we have to remove that fee from the holdings

Personally, I would prefer to see for each trade the fee in a separate column, and not included in the rate and trade amounts.
